### PR TITLE
power complex division by zero

### PR DESCRIPTION
**Pull request by alinioan**
Fixed case where pow for complex numbers doesn't result in division by zero. (https://github.com/RustPython/RustPython/issues/5138)
pow(0j, -1) now results in a division by zero error.

Added unit test for this condition.

### CODE REVIEW

File: Lib/test/test_complex.py [hunk 1]
```diff
@@ -241,6 +241,13 @@ def test_pow(self):
         else:
             self.fail("should fail 0.0 to negative or complex power")
 
+        try:
+            0j ** (-1)
+        except ZeroDivisionError:
+            pass
+        else:
+            self.fail("should fail 0.0 to negative or complex power")
```
Comment (youknowone) [inline Lib/test/test_complex.py:249]: Could you revert this change? Since the test file is originated from CPython project, we usually don't edit it ourselves.

File: vm/src/builtins/complex.rs [hunk 2]
```diff
@@ -103,7 +103,7 @@ fn inner_div(v1: Complex64, v2: Complex64, vm: &VirtualMachine) -> PyResult<Comp
 
 fn inner_pow(v1: Complex64, v2: Complex64, vm: &VirtualMachine) -> PyResult<Complex64> {
     if v1.is_zero() {
-        return if v2.im != 0.0 {
+        return if v2.im != 0.0 || (v1.im.is_zero() && v2.re().is_negative()) {
```
Comment (youknowone) [inline vm/src/builtins/complex.rs:106]: The patch seems not to be based on the recent upstream.

#5143 has similar changes

**Review by youknowone — CHANGES_REQUESTED** [review]

File: vm/src/builtins/complex.rs (hunk 2 above)
Comment (alinioan) [reply vm/src/builtins/complex.rs:106]: Yes, sorry about this. When i first fixed this error it was from [this issue](https://github.com/RustPython/RustPython/issues/5138) which wasn't closed.
